trie/rootManager.go

```diff
@@ -0,0 +1,75 @@
+package trie
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"sync"
+)
+
+type rootManager struct {
+	root         node
+	oldHashes    [][]byte
+	oldRootHash  []byte
+	mutOperation sync.RWMutex
+}
+
+func NewRootManager() *rootManager {
```

Comment:
a lot of comments are missing here. on all the file.

Reply:
Added